### PR TITLE
Add Idle Timeout for Load Balancers (#44)

### LINKED ISSUE #44: Idle Timeout for Elastic Load Balancers should be configurable
When using Helm/Tiller, deployments make take longer than the 60 second default timeout when communicating with the API Server.  While the precise issue is related to the API Server, it is logical to allow configuration of all ELBs.

### PR DESCRIPTION
**Pull request by hobti01**
Needed in support of load balancer idle timeout configuration within https://github.com/giantswarm/aws-operator/pull/446

### CODE REVIEW

**Comment by hobti01** [comment]
Good point @puja108 , I'll add that logic into giantswarm/aws-operator#446

File: spec/aws/elb.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package aws
+
+import "github.com/giantswarm/awstpr/spec/aws/elb"
+
+// ELB contains additional settings for the elastic load balancers.
+type ELB struct {
+	// IdleTimeoutSeconds is idle time before closing the front-end and back-end connections
+	IdleTimeoutSeconds elb.AWS `json:"idleTimeoutSeconds" yaml:"idleTimeoutSeconds"`
+}
```
Comment (rossf7) [inline spec/aws/elb.go]: I think this should be.

```go
type ELB struct {
	API     elb.ELB `json:"api" yaml:"api"`
	Etcd    elb.ELB `json:"etcd" yaml:"etcd"`
	Ingress elb.ELB `json:"ingress" yaml:"ingress"`
}
```

File: spec/aws/elb/idle_timeout_seconds.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+// AWS contains the Idle Timeout in seconds of the elastic load balancers.
+type AWS struct {
+	// API is the Idle Timeout in seconds for the Kubernetes API Server load balancer.
+	API int `json:"api" yaml:"api"`
+	// Etcd is the Idle Timeout in seconds for the etcd load balancer.
+	Etcd int `json:"etcd" yaml:"etcd"`
+	// Ingress is the Idle Timeout in seconds for the Ingress load balancer, used for customer traffic.
+	Ingress int `json:"ingress" yaml:"ingress"`
+}
```
Comment (rossf7) [inline spec/aws/elb/idle_timeout_seconds.go]: Here it should be.

```go
type ELB struct {
	API     elb.ELB `json:"api" yaml:"api"`
	Etcd    elb.ELB `json:"etcd" yaml:"etcd"`
	Ingress elb.ELB `json:"ingress" yaml:"ingress"`
}
```

Comment (hobti01) [reply spec/aws/elb/idle_timeout_seconds.go]: Hi Ross,  the yaml structure is like this:
```
  aws:
    region: "{{.Values.aws.region}}"
    az: "{{.Values.aws.az}}"
    elb:
      idleTimeoutSeconds:
        api: 120
        etcd: 60
        ingress: 60
```
So these field should be `int` unless you prefer a different yaml structure.

The name of the IdleTimeoutSeconds struct has a typo that I will change (AWS -> IdleTimeoutSeconds)

I'm conflicted with either the above structure, or using:
```
  aws:
    region: "{{.Values.aws.region}}"
    az: "{{.Values.aws.az}}"
    elb:
      api:
        idleTimeoutSeconds: 120
      etcd:
        idleTimeoutSeconds: 120
      ingress:
        idleTimeoutSeconds: 120
```
Do you have a preference?

**Comment by hobti01** [comment]
Thanks for the pointer on the tests, I wasn't sure how you wanted that.  I'll update.

File: spec/aws/elb/idle_timeout_seconds.go (hunk 2 above)
Comment (rossf7) [reply spec/aws/elb/idle_timeout_seconds.go]: Ah yes I see what you mean! Let's go with the first format. It's less verbose but still easy to read.

**Comment by hobti01** [comment]
@puja108 or @rossf7 - would you prefer than a negative timeout value is silently set to the default or that it fails validation?

The maximum value is being checked via the `validateCluster` function.

**Comment by rossf7** [comment]
Hi Tim, I'd prefer setting to the default for negative values.

**Comment by hobti01** [comment]
@rossf7 please review the tests and other updates

**Pull request merged**